src/main/java/org/junit/internal/runners/statements/RunBefores.java

```diff
@@ -6,13 +6,24 @@
 import org.junit.runners.model.Statement;
 
 public class RunBefores extends Statement {
+
+    private final Statement previous;
+
     private final Statement next;
 
     private final Object target;
 
     private final List<FrameworkMethod> befores;
 
-    public RunBefores(Statement next, List<FrameworkMethod> befores, Object target) {
+    /**
+     * @param previous The previous statement, or {@code null}.
+     * @param next The next statement after before methods.
+     * @param befores The list of before methods.
+     * @param target
+     */
+    public RunBefores(final Statement previous, final Statement next,
```

Comment:
Unfortunately, `RunBefores` is a public class. Even though it's in an internal package, we should keep the previous constructor around (though we could deprecate it)